File: src/main/java/com/antero/liemipeli/domain/paikat/Varasto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.antero.liemipeli.domain.paikat;

import com.antero.liemipeli.domain.Tuote;
import com.antero.liemipeli.domain.TuoteImpl;
import java.util.ArrayList;
import java.util.Collection;

/**
 *
 * @author Antero Oikkonen
 */
public class Varasto implements VarastoInterface {

    private Collection<Tuote> tuotteet = new ArrayList();
    private int koko;

    public Varasto(int koko) {
        this.koko = koko;
    }

    @Override
    public Tuote haeTuote(String nimi) {
        for (Tuote t : tuotteet) {
            if (t.getNimi().equals(nimi)) {
                tuotteet.remove(t);
                return t;
            }
        }
        return null;
    }

    @Override
    public boolean lisaaTuote(Tuote tuote) {
        if (tuotteet.size() == this.koko) {
            return false;
        }
        return tuotteet.add(tuote);
    }

    @Override
    public int asetaVarastonKoko(int koko) {
        if (koko < 0) {
            throw new IllegalArgumentException("koko pitää olla positiivinen");
        }
        if (koko < tuotteet.size()) {
            koko = tuotteet.size();
        }
        this.koko = koko;
        return koko;
    }

    @Override
    public int laskeVarastonArvo() {
        int arvo = 0;
        for (Tuote t : tuotteet) {
            arvo += t.getHinta();
        }
        return arvo;
    }

}

File: liemipeli-kotlin/src/main/kotlin/com/antero/liemipeli/domain/paikat/Varasto.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.antero.liemipeli.domain.paikat

import com.antero.liemipeli.domain.Tuote
import java.util.*

/**
 *
 * @author Antero Oikkonen
 */
class Varasto(private var koko: Int) : VarastoInterface {

    private val tuotteet = ArrayList<Tuote>()


    override fun haeTuote(nimi: String): Tuote? {
        for (t in tuotteet) {
            if (t.nimi == nimi) {
                tuotteet.remove(t)
                return t
            }
        }
        return null
    }

    override fun lisaaTuote(tuote: Tuote): Boolean {
        return if (tuotteet.size == this.koko) {
            false
        } else tuotteet.add(tuote)
    }

    override fun asetaVarastonKoko(vkoko: Int): Int {

        var tempKoko = vkoko

        if (vkoko < 0) {
            throw IllegalArgumentException("koko pitää olla positiivinen")
        }
        if (vkoko < tuotteet.size) {
            tempKoko = tuotteet.size
        }
        this.koko = tempKoko
        return this.koko
    }

    /**
     * mikä on varaston arvo
     */
    override fun laskeVarastonArvo(): Int {
        var arvo = 0
        for (t in tuotteet) {
            arvo += t.hinta
        }
        return arvo
    }

    /**
     * monta tuotetta on varastossa
     * @return
     */

    override fun varastoSaldo(): Int {
        return tuotteet.size
    }

}
